.github/workflows/phpunit.yml

```diff
@@ -0,0 +1,83 @@
+name: PHPUnit
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - stable*
+
+jobs:
+  php:
+    runs-on: ubuntu-latest
+
+    strategy:
+      # do not stop on another job's failure
+      fail-fast: false
+      matrix:
+        php-versions: ['7.2', '7.3', '7.4']
+        databases: ['sqlite', 'mysql']
+        server-versions: ['master']
+  
+    name: php${{ matrix.php-versions }}/${{ matrix.databases }} on ${{ matrix.server-versions }}
+
+    services:
+      mariadb:
+        image: mariadb:latest
+        ports:
+          - 3306
+        env:
+          MYSQL_USER: nextcloud
+          MYSQL_PASSWORD: password
+          MYSQL_DATABASE: nextcloud
+          MYSQL_ROOT_PASSWORD: rootpassword
+        options: --health-cmd="mysqladmin ping" --health-interval=5s --health-timeout=2s --health-retries=3
+      # postgres:
+      #   image: postgres
+      #   env:
+      #     POSTGRES_USER: nextcloud
+      #     POSTGRES_PASSWORD: password
+      #     POSTGRES_DB: nextcloud
+      #   ports:
+      #     - 5432/tcp
+
+    steps:
+      - name: Checkout server
+        uses: actions/checkout@v2
+        with:
+          repository: nextcloud/server
+          ref: ${{ matrix.server-versions }}
+      - name: Checkout submodules
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+      - name: Checkout app
+        uses: actions/checkout@v2
+        with:
+          path: apps/notestutorial
+      - name: Set up php${{ matrix.php-versions }}
+        uses: shivammathur/setup-php@v1
+        with:
+          php-version: ${{ matrix.php-versions }}
+          tools: phpunit
+          extensions: intl, pdo_mysql, pdo_pgsql
+          coverage: none
+      - name: Set up nextcloud
+        env:
+          # get the random port from the service
+          DB_PORT: ${{ job.services.mariadb.ports['3306'] }}
+        run: |
+          pwd
+          ls -la
+          mkdir data
+          ./occ maintenance:install --verbose --database=${{ matrix.databases }} --database-name=nextcloud --database-host=127.0.0.1 --database-port=$DB_PORT --database-user=root --database-pass=rootpassword --admin-user admin --admin-pass password 
```

Comment:
@ChristophWurst @rullzer  @juliushaertl if you have any clue here, it cannot find the server.
https://github.community/t5/GitHub-Actions/github-actions-cannot-connect-to-mysql-service/td-p/30611

tried various solutions, never worked. Tried accessing the service via the service name as host, doesn't work as well... The occ is not helpful.
Though it seems to fail with `Error while trying to create admin user: Failed to connect to the database: An exception occurred in driver: SQLSTATE[HY000] [1045] Access denied for user 'oc_admin'@'localhost' (using password: YES)` which seems to say, since I'm loging it as root on the job, that it created its own user, right? Any ideas?

Reply:
> 127.0.0.1

isn't it running in another container?

Reply:
No, everything is on the same host. This is weird stuff.
I'll add a manual db connect step, you'll see
https://firefart.at/post/using-mysql-service-with-github-actions/

Reply:
![image](https://user-images.githubusercontent.com/14975046/73675067-104fa700-46b2-11ea-94a4-74c3b067eac7.png)

Reply:
no the service is a separate container and should be available under the service name `mariadb` as hostname

Reply:
Ah you can also use `${{ job.services.mysql.ports[3306] }}` to get the actual mysql port

https://github.com/actions/example-services/blob/master/.github/workflows/postgres-service.yml#L44-L72

Reply:
Yes that is what I'm using

Reply:
Tried other stuff, I'm running out of ideas :(